### PR TITLE
has: init at 1.4.0

### PR DESCRIPTION
**Pull request by Freed-Wu**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Freed-Wu** [comment]
Any problem?

**Comment by Freed-Wu** [comment]
Can anyone review this PR? Thanks!

File: pkgs/applications/misc/has/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:3]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

File: pkgs/applications/misc/has/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "has";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "kdabir";
+    repo = "has";
+    rev = "v" + version;
+    sha256 = "1482k7mz2crh2i00zcll0f9wc580z9ca0ipk21iz4h95br50syyx";
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:11]: use `hash` instead of `sha256`.

File: pkgs/applications/misc/has/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm0555 ${pname} -t $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/kdabir/has";
+    description = "checks presence of various command line tools and their versions on the path ";
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:24]: Please capitalize the first letter and remove the trailing empty space.

File: pkgs/applications/misc/has/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:26]: Please consider adding yourself as maintainer.

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by Freed-Wu** [comment]
Fixed.

File: pkgs/applications/misc/has/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "has";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "kdabir";
+    repo = "has";
+    rev = "v" + finalAttrs.version;
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:10]:
```suggestion
    rev = "v${finalAttrs.version}";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by Freed-Wu** [comment]
Fixed.

File: pkgs/applications/misc/has/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "has";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "kdabir";
+    repo = "has";
+    rev = "v${finalAttrs.version}";
+    sha256 = "";
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:11]:
```suggestion
    hash = "<something here>";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/has/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub }:
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, fetchFromGitHub }:
```

File: pkgs/applications/misc/has/default.nix [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
```
Comment (drupol) [inline pkgs/applications/misc/has/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by Freed-Wu** [comment]
Fixed.

**Review by drupol — APPROVED** [review]

**Pull request merged**